tasks/preflight_cpu_mem_checks.yml

```
@@ -0,0 +1,118 @@
+---
+- name: Check CPU supports x86-64-v2 for RHEL 9
+  shell: "lscpu | grep -q 'avx2' && echo 'yes' || echo 'no'"
+  register: cpu_supports_x86_64_v2
+  changed_when: false
+
+- name: Fail if CPU does not support x86-64-v2
+  fail:
+    msg: "CPU does not support x86-64-v2. RHEL 9 requires at least AVX2 support."
+  when: cpu_supports_x86_64_v2.stdout != "yes"
+  register: cpu_fail
+
+- name: Store CPU instruction set check result
+  set_fact:
+    preflight_results: "{{ preflight_results + [
+      {
+        'Check': 'CPU x86-64-v2',
+        'Result': 'PASS' if cpu_supports_x86_64_v2.stdout == 'yes' else 'FAIL',
+        'Reason': 'AVX2 instruction set missing' if cpu_supports_x86_64_v2.stdout != 'yes' else 'N/A'
+      }
+    ] }}"
+
+- name: Check available CPU cores
+  shell: "nproc"
```

no need to use a `shell` task here
```suggestion
  ansible.builtin.command: "nproc"
```

---

by the way, that's probably something you want to get from ansible facts :

```
        "processor": [
            "0",
            "AuthenticAMD",
            "AMD EPYC-Rome Processor",                                                                                                                                                                                                                                                                                                                                                                                                                  "1",
            "AuthenticAMD",
            "AMD EPYC-Rome Processor",
            "2",
            "AuthenticAMD",
            "AMD EPYC-Rome Processor",
            "3",
            "AuthenticAMD",
            "AMD EPYC-Rome Processor"
        ],
        "processor_cores": 1,
        "processor_count": 4,
        "processor_nproc": 4,
        "processor_threads_per_core": 1,
        "processor_vcpus": 4,
```

---

Addressed